server/src/models/internship.ts

```diff
@@ -139,6 +139,8 @@ const requiredPdfs = [
 
 function internshipRequestComplete(document: Document) {
   // Check if all required fields exist
+  if (document.get("status") !== InternshipStatuses.PLANNED) return false;
```

Comment:
Wenn hier `false` zurückgegeben wird, dann setzt `trySetRequested` den Status doch zurück auf `PLANNED`, sobald die Methode nochmal aufgerufen wird, oder?

Reply:
da hast du allerdings recht! ich denke man sollte den fall abfangen dass wenn es requested ist, dass man es dann nicht nochmal auf requestet setzt. aber wo? hier oder in trySetRequested?

Reply:
Ich würde sagen in `trySetRequested`. Da können wir dann schauen, ob es vollständig ist und in dem Fall den Status auf `REQUESTED` setzen, aber nur wenn der Status vorher `PLANNED` war. Ansonsten bleibt er einfach so, wie er ist. Und falls es nicht vollständig ist und der Status nicht `PLANNED` ist, dann wird's wieder auf `PLANNED` gesetzt.